src/engine/bufferscalers/enginebufferscalerubberband.cpp

```diff
@@ -28,22 +26,16 @@ size_t kRubberBandBlockSize = 256;
 EngineBufferScaleRubberBand::EngineBufferScaleRubberBand(
         ReadAheadManager* pReadAheadManager)
         : m_pReadAheadManager(pReadAheadManager),
-          m_buffer_back(SampleUtil::alloc(MAX_BUFFER_LEN)),
+          m_buffers{std::vector<CSAMPLE>(MAX_BUFFER_LEN), std::vector<CSAMPLE>(MAX_BUFFER_LEN)},
+          m_bufferPtrs{m_buffers[0].data(), m_buffers[1].data()},
+          m_interleavedReadBuffer(MAX_BUFFER_LEN),
```

Comment:
also replacing a SampleUtil::alloc(MAX_BUFFER_LEN) via a std::vector<CSAMPLE>(MAX_BUFFER_LEN) is not exactly equivalent as far as I know. `SampleUtil::alloc` allocates aligned to the SIMD lane boundaries to aid vectorization (and the performance of the generated code). Also the manual allocation guaranteed that there was no accidental realloc which might be the case with the vector (if it decides to silently grow due to programmer error). I'm generally in favor of using std containers and algorithms so I approve of using `std::span` whenever possible and appropriate. Due to the possibility of `std::vector` to allocate, I'm not comfortable seeing it in realtime code.

We have an RAII type that does have the semantics we are looking for: `SampleBuffer` (in `util/samplebuffer.h`). Then use the `SampleBuffer::span() -> std::span<CSAMPLE>` method (instead of `size`/`data`/`ReadableSlice`/`WriteableSlice` methods and classes).

Reply:
Done.